spec/unit_tests/moab/storage_object_validator_spec.rb

```diff
@@ -15,104 +15,155 @@
+          expect(verification_array[7]).to eq(Moab::StorageObjectValidator::METADATA_SUB_DIRS_DETECTED =>"Should only contain files, but directories were present in the metadata directory")
+        end
+        it 'Content does not have any files present' do
+          # v0009
+          expect(verification_array[8]).to eq(Moab::StorageObjectValidator::NO_FILES_IN_CONTENT_DIR=>"Version:  No files present in content dir")
+        end
+        it 'Metadata does not have any files present' do
+          # v0010
+          expect(verification_array[9]).to eq(Moab::StorageObjectValidator::NO_FILES_IN_METADATA_DIR=>"Version:  No files present in metadata dir")
+        end
+      end
+      context 'under manifest directory' do
+        druid = 'cc000cc0000'
+        druid_path = 'spec/fixtures/bad_root01/bad_moab_storage_trunk/cc/000/cc/0000/cc000cc0000'
+        storage_obj = Moab::StorageObject.new(druid, druid_path)
+        storage_obj_validator = described_class.new(storage_obj)
+        verification_array = storage_obj_validator.validation_errors
+        it 'does not have signatureCatalog.xml' do
+          expect(verification_array).to include(Moab::StorageObjectValidator::NO_SIGNATURE_CATALOG => "Version: v0001 Missing signatureCatalog.xml")
+        end
+
+        it 'does not have manifestInventory.xml' do
+          expect(verification_array).to include(Moab::StorageObjectValidator::NO_MANIFEST_INVENTORY => "Version: v0002 Missing manifestInventory.xml")
+        end
+
+        it 'does not have either signatureCatalog.xml or manifestInventory.xml' do
+          expect(verification_array).to include(Moab::StorageObjectValidator::NO_FILES_IN_MANIFEST_DIR => "Version: v0003 No files present in manifest dir")
+        end
+      end
+      it "has non contiguous version directories" do
+        druid = 'yy000yy0000'
+        druid_path = 'spec/fixtures/bad_root01/bad_moab_storage_trunk/yy/000/yy/0000/yy000yy0000'
+        storage_obj = Moab::StorageObject.new(druid, druid_path)
+        storage_obj_validator = described_class.new(storage_obj)
+        verification_array = storage_obj_validator.validation_errors
+        expect(verification_array).to include(Moab::StorageObjectValidator::VERSIONS_NOT_IN_ORDER =>"Should contain only sequential version directories. Current directories: [\"v0001\", \"v0003\", \"v0004\", \"v0006\"]")
+      end
+      it "has extra characters in version directory name" do
+        druid = 'aa000aa0000'
+        druid_path = 'spec/fixtures/bad_root01/bad_moab_storage_trunk/aa/000/aa/0000/aa000aa0000'
+        storage_obj = Moab::StorageObject.new(druid, druid_path)
+        storage_obj_validator = described_class.new(storage_obj)
+        verification_array = storage_obj_validator.validation_errors
+        expect(verification_array).to include(Moab::StorageObjectValidator::VERSION_DIR_BAD_FORMAT => "Version directory name not in 'v00xx' format")
+      end
+      it "incorrect items in path" do
```

Comment:
i don't understand this error message, sadly -- make a separate ticket to clarify this (patch party).  Is it that unexpected files/dirs are found under the xxxxx (version??) directory?  Under the moab directory?

Reply:
ticket #97 created!

Reply:
agreed, a little unclear.  looks from the file structure of the test fixture like it's unexpected files under the version dir:
```
├── bb
│   └── 000
│       └── bb
│           └── 0000
│               └── bb000bb0000
│                   └── v0001
│                       ├── test1.txt
│                       └── test2.txt
```